### PR TITLE
chore: refactor the frontends chain list usage (LF-4516)

### PR DESCRIPTION
**Pull request by dennyscode**
Ticket: https://lifi.atlassian.net/browse/LF-4516

### CODE REVIEW

File: packages/dapp/src/types/internal.types.ts [hunk 1]
```diff
@@ -64,7 +55,7 @@ export interface ChainsMenuListItem {
   showMoreIcon?: boolean;
   checkIcon?: boolean;
   onClick?: any;
-  chainId: ChainId;
+  chainId: number;
```
Comment (dennyscode) [inline packages/dapp/src/types/internal.types.ts:58]: Is this legit? Still not sure about it

File: packages/shared/src/types/multisig.ts [hunk 2]
```diff
@@ -1,7 +1,5 @@
-import type { ChainId } from '@lifi/types';
-
 export interface MultisigProps {
-  destinationChain?: ChainId;
+  destinationChain?: number;
```
Comment (dennyscode) [inline packages/shared/src/types/multisig.ts:2]: Same here

File: packages/dapp/src/types/internal.types.ts (hunk 1 above)
Comment (dennyscode) [reply packages/dapp/src/types/internal.types.ts:58]: Okay, you said this multiple times, thanks for you patience, will bring this back!

File: packages/dapp/src/types/internal.types.ts [hunk 3]
```diff
@@ -79,87 +71,6 @@ export interface ChainsMenuListItem {
...
-export interface WalletAccount {
-  address?: string;
-  isActive?: boolean;
-  signer?: Signer;
-  chainId?: number;
-}
-
 export type MultisigWidgetConfig = Pick<
   WidgetConfig,
   'fromChain' | 'requiredUI'
```
Comment (dennyscode) [inline packages/dapp/src/types/internal.types.ts:76]: All of that was junk not in use. 
Wallet related stuff was duplicated and should be used from wallet.ts instead, so got rid of what was found in here.

**Review by chybisov — CHANGES_REQUESTED** [review]

File: packages/dapp/src/types/userTracking.types.ts [hunk 4]
```diff
@@ -1,5 +1,5 @@
 import { ChainID } from '@arcxmoney/analytics';
-import { WalletAccount } from './internal.types';
+import { WalletAccount } from '@transferto/shared/src/types';
```
[the remark this answers is not in the document]
Comment (dennyscode) [reply packages/dapp/src/types/userTracking.types.ts:2]: Created this task:
https://lifi.atlassian.net/browse/LF-4923

File: packages/dapp/src/components/Navbar/Menu/WalletMenu/WalletMenu.tsx [hunk 5]
```diff
@@ -17,6 +16,7 @@ import { SpotButton } from '@transferto/shared/src/atoms';
 import { openInNewTab, walletDigest } from '@transferto/shared/src/utils';
 import { useCallback, useEffect, useMemo, useState } from 'react';
 import { useTranslation } from 'react-i18next';
+import { useChains } from '../../../../hooks/useChains';
```
Comment (chybisov) [inline packages/dapp/src/components/Navbar/Menu/WalletMenu/WalletMenu.tsx:19]: Also, you know there is something wrong when you need to make `../../../../` imports 😄

**Review by chybisov — APPROVED** [review]

**Review by Abhikumar98 — APPROVED** [review]

**Pull request merged**